### PR TITLE
refactor/migrate-to-doks

### PR DESCRIPTION
**Pull request by DannyS03**
## Context 

**Migrates docs.libp2p.io from the Hugo Learn theme to the Hugo Doks theme.**
- Landing page will better designed and updated in follow-up PR.
- Will request some icons and libp2p-based graphics for the docs site from Crcle, including a cleaner "docs logo".
- "Notices" turned into comments for now.
- Tabs to be readded when upgrading to [Docs-starter](https://github.com/protocol/docs-starter).
- CI (+ lint checker) and README updated to reflect config.
  - ~Need access to Fleek account to update.~
- While going through the docs again, I came across a lot of opportunity to update the current material which we'll do so right after.
- Includes necessary redirects and link updates.
- ~Top-level "Guides" section to be when we include more guides.~
- Proposing to have reference on the Menu/Navbar as the current reference material just includes references.

## Latest preview

Please view the latest Fleek preview [here](https://libp2p-docs-beta.on.fleek.co).

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Hugo build was successful!
- No markdown files were changed, so no page checks were run!

**Comment by filecorgi** [comment]
- I optimized some images for you! See the commit with the comment `Automatically optimized images [ci skip].` in this PR for details.
- Hugo build was successful!

**Comment by filecorgi** [comment]
- I optimized some images for you! See the commit with the comment `Automatically optimized images [ci skip].` in this PR for details.
- Hugo build failed...

```
Installing Hugo to ./bin/hugo...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

 10 16.2M   10 1743k    0     0  2553k      0  0:00:06 --:--:--  0:00:06 2553k
100 16.2M  100 16.2M    0     0  14.4M      0  0:00:01  0:00:01 --:--:-- 32.9M
mv: cannot move 'tmp_hugo/hugo' to './bin/hugo': No such file or directory
make: *** [Makefile:42: bin/hugo] Error 1
```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Hugo build was successful!

**Comment by filecorgi** [comment]
- I optimized some images for you! See the commit with the comment `Automatically optimized images [ci skip].` in this PR for details.
- Hugo build was successful!

**Comment by p-shahi** [comment]
Looks like the fleek preview is broken, as you suggested in DMs, we could have a beta site for this migration

File: content/en/versions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: "Versions"
```
Comment (p-shahi) [inline content/en/versions.md:2]: do we need this file?

File: data/docs-versions.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# - group: v1.x
```
Comment (p-shahi) [inline data/docs-versions.yml:1]: do we need this file?

**Review by p-shahi — CHANGES_REQUESTED** [review]
Also think we should delete the static/videos directory

File: deprecated/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
+<p align="center">
```
[the remark this answers is not in the document]
Comment (DannyS03) [reply deprecated/README.md:1]: I'd recommend we keep it as "deprecated" or "archive" for a period of time, what do you think?

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Contributor Covenant Code of Conduct
```
[the remark this answers is not in the document]
Comment (DannyS03) [reply CODE_OF_CONDUCT.md:1]: It's from [here](https://www.contributor-covenant.org/) and was just part of the Doks projects.

File: content/en/docs/appendix/_index.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+title : "Appendix"
+description: "Stream Multiplexing is a way of sending multiple streams of data over one communication link. It combines multiple signals into one unified signal so it can be transported 'over the wires', then it is demulitiplexed so it can be output and used by separate applications."
+weight: 12
+aliases: /concepts/stream-multiplexing/
```
[the remark this answers is not in the document]
Comment (DannyS03) [reply content/en/docs/appendix/_index.md:5]: general =)

File: deprecated/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (p-shahi) [reply deprecated/README.md:1]: I don't see why we need to keep it since it'll be preserved in the commit history.

**Comment by p-shahi** [comment]
I missed this in my last review but this PR still contains some leftover assets from the Doks like images/doks.png,  images/screenshot.png, and images/tn.png

File: netlify.toml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+[build]
```
Comment (p-shahi) [inline netlify.toml]: We should remove this since we don't use Netlify

File: i18n/de.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+- id: get-started
```
Comment (p-shahi) [inline i18n/de.yaml:1]: Can we remove de.yaml and nl.yaml

File: config/_default/languages.toml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  contentDir = "content/nl"
+  weight = 20
+  [nl.params]
+    languageISO = "NL"
+    languageTag = "nl-NL"
+    titleAddition = "Modern documentatie-thema"
+    description = "Doks is een Hugo-thema waarmee je moderne documentatie-websites kunt bouwen die veilig, snel en klaar voor SEO zijn — standaard."
+    titleHome = "Doks thema"
+    footer = "Mogelijk gemaakt door <a href=\"https://www.netlify.com/\">Netlify</a>, <a href=\"https://gohugo.io/\">Hugo</a>, en <a href=\"https://getdoks.org/\">Doks</a>"
+    alertText = "Introductie van het Doks-kinderthema, verschillende DX + UX-updates en meer! <a class=\"alert-link stretched-link\" href=\"https://getdoks.org/blog/doks-v0.2/\">Bekijk Doks v0.2</a>"
```
Comment (p-shahi) [inline config/_default/languages.toml:28]:
```suggestion
```

File: CODE_OF_CONDUCT.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (p-shahi) [reply CODE_OF_CONDUCT.md:1]: I'd remove it and link to https://github.com/ipfs/community/blob/master/code-of-conduct.md similar to https://github.com/libp2p/js-libp2p/blob/master/CODE_OF_CONDUCT.md

File: netlify.toml (hunk 6 above)
Comment (DannyS03) [reply netlify.toml]: Yep noticed that too and removed it.

File: i18n/de.yaml (hunk 7 above)
Comment (DannyS03) [reply i18n/de.yaml:1]: Should we remove the whole i18n until we need it?

Comment (p-shahi) [reply i18n/de.yaml:1]: Yeah sounds good

**Comment by p-shahi** [comment]
I see the preview is here now: https://libp2p-docs-beta.on.fleek.co/ 
Looking good, will review it too

File: content/en/docs/security/dos-mitigation.md [hunk 9]
```diff
@@ -0,0 +1,408 @@
+---
+title : "DoS Mitigation"
```
Comment (p-shahi) [inline content/en/docs/security/dos-mitigation.md:2]: content/en/docs/dos-mitigation.md

DoS mitigation should be it's own section like how it is here: https://docs.libp2p.io/concepts/dos-mitigation/

File: config/_default/languages.toml [hunk 10]
```diff
@@ -0,0 +1,28 @@
+[en]
+  languageName = "English"
+  contentDir = "content/en"
```
Comment (p-shahi) [inline config/_default/languages.toml:3]:
```suggestion
  contentDir = "content/"
```
Can we keep the directory structure for the content the same?
That would make this PR much smaller and let's us keep the file history for the actual docs.

Basically reverting changes like this:
[content/concepts/fundamentals/addressing.md → content/en/docs/fundamentals/addressing.md](https://github.com/libp2p/docs/pull/220/files#diff-e69ee054730b149e391f572e7b10dce989535ff450796f0c7aa1135b2a425f7e)

File: config/_default/menus/menus.en.toml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  name = "Reference"
+  weight = 20
+  identifier = "reference"
+
+[[main]]
+  name = "Specifications"
+  weight = 10
+  identifier = "quick-start"
+  url = "https://github.com/libp2p/specs/"
+  parent = "reference"
```
Comment (p-shahi) [inline config/_default/menus/menus.en.toml:16]:
```suggestion
[[reference]]
  name = "Specifications"
  weight = 10
  identifier = "specifications"
  url = "https://github.com/libp2p/specs/"
```

File: config/_default/menus/menus.en.toml [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+[[main]]
+  name = "Connectivity"
+  weight = 20
+  identifier = "tutorial"
+  url = "https://connectivity.libp2p.io/"
+  parent = "reference"
```
Comment (p-shahi) [inline config/_default/menus/menus.en.toml:23]:
```suggestion
[[reference]]
  name = "Connectivity"
  weight = 20
  identifier = "connectivity"
  url = "https://connectivity.libp2p.io/"
```

File: config/_default/menus/menus.en.toml [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+[[main]]
+  name = "Implementations"
+  weight = 30
+  identifier = "implementations"
+  url = "https://libp2p.io/implementations/"
+  parent = "reference"
```
Comment (p-shahi) [inline config/_default/menus/menus.en.toml:30]:
```suggestion
[[reference]]
  name = "Implementations"
  weight = 30
  identifier = "implementations"
  url = "https://libp2p.io/implementations/"
```

File: config/_default/menus/menus.en.toml [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  name = "Forum"
+  url = "https://discuss.libp2p.io/"
+  weight = 10
+  parent = "community"
+
+[[main]]
+  name = "Blog"
+  url = "/blog/"
+  weight = 20
+  parent = "community"
```
Comment (p-shahi) [inline config/_default/menus/menus.en.toml:48]:
```suggestion
[[main]]
  name = "Community"
  url = "/community/"
  weight = 40
  identifier = "community"

[[community]]
  name = "Get Involved"
  url = "https://libp2p.io/#community"
  weight = 10

[[community]]
  name = "Forum"
  url = "https://discuss.libp2p.io/"
  weight = 10

[[main]]
  name = "Blog"
  url = "/blog/"
  weight = 20
```

File: config/_default/menus/menus.en.toml [hunk 15]
```diff
@@ -0,0 +1,66 @@
+[[main]]
+  name = "Docs"
+  url = "/docs/introduction/overview"
+  weight = 10
+
+[[main]]
+  name = "Reference"
+  weight = 20
+  identifier = "reference"
+
```
Comment (p-shahi) [inline config/_default/menus/menus.en.toml:10]:
```suggestion

[[main]]
  name = "Guides"
    url = "/docs/guides"
  weight = 20
  identifier = "guides"
```

Let's move the guides and tutorials (content in https://libp2p-docs-beta.on.fleek.co/docs/guides/go/) out to a different tab like https://docs.filecoin.io/get-started/overview/

File: archetypes/docs/lorem/_index.md [hunk 16]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Lorem"
```
Comment (p-shahi) [inline archetypes/docs/lorem/_index.md:2]: Let's delete this, archetypes/docs/lorem/ipsum/index.md, assets/images/default-image.png, and any empty directory with a .gitkeep file

**Review by p-shahi — COMMENTED** [review]
Noticed the favicons aren't right, we should keep https://github.com/libp2p/docs/blob/master/static/images/favicon-32x32.png in place of static/favicon-32x32.png (and other favicon files)

File: content/en/docs/security/dos-mitigation.md (hunk 9 above)
Comment (DannyS03) [reply content/en/docs/security/dos-mitigation.md]: @p-shahi I collapsed the menu (I believe it is cleaner like this) which shows DoS mitigation without an additional click. Does this work?

**Review by p-shahi — APPROVED** [review]
🍻 I think this looks awesome, thank you so much @DannyS03 
I suggest sharing this link https://libp2p-docs-beta.on.fleek.co with the team to get any more feedback before merging.

Also, I've seen the fleek/build succeed every now and then. Let's resolve that and the libp2p-docs before merging

File: content/en/docs/security/dos-mitigation.md (hunk 9 above)
Comment (p-shahi) [reply content/en/docs/security/dos-mitigation.md]: Sounds good!

**Comment by DannyS03** [comment]
The main site build is now updated with the same configuration as the beta site.

**Pull request merged**